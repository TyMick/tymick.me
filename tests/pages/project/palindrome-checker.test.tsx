import React from "react";
import renderer from "react-test-renderer";
import PalindromeChecker from "../../../pages/project/palindrome-checker";

jest.mock("react-gist");

describe("PalindromeChecker", () => {
  it("matches snapshot", async () => {
    expect(renderer.create(<PalindromeChecker />).toJSON()).toMatchSnapshot();
  });
});
