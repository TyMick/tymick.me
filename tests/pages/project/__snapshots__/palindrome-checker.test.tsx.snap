// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PalindromeChecker matches snapshot 1`] = `
<div
  className="container"
>
  <h1
    className="mb-4"
  >
    Palindrome checker algorithm
  </h1>
  <div
    className="mx-2 mx-sm-5 mb-3"
  >
    <form
      className="mb-3 text-monospace form-inline"
    >
      <div
        className="w-100 input-group"
      >
        <div
          className="input-group-prepend"
        >
          <span
            className="input-group-text"
            id="functionName"
          >
            palindrome
            (
          </span>
        </div>
        <input
          aria-describedby="functionName"
          aria-label="str"
          className="form-control"
          onChange={[Function]}
          placeholder="str"
          type="text"
        />
        <div
          className="input-group-append"
        >
          <span
            className="input-group-text"
          >
            )
          </span>
        </div>
      </div>
    </form>
    <div
      className="d-flex align-items-center"
    >
      <h5
        className="mb-0 mr-3"
      >
        Returns:
      </h5>
      <div
        className="h5 mb-0 card"
      >
        <pre
          className="px-2 py-1 m-0"
        >
          Â 
        </pre>
      </div>
    </div>
  </div>
  <div
    className="mb-2"
  />
  <p>
    I created this algorithm as a requirement for
     
    <a
      href="https://www.freecodecamp.org/certification/tymick/javascript-algorithms-and-data-structures"
    >
      my freeCodeCamp JavaScript Algorithms and Data Structures Certification
    </a>
    . It returns 
    <code>
      true
    </code>
     if the given string is a palindrome, ignoring letter case and all non-alphanumeric characters. Otherwise, it returns 
    <code>
      false
    </code>
    .
  </p>
</div>
`;
